common/config/networks.ts

```diff
@@ -0,0 +1,332 @@
+import { ethPlorer, ETHTokenExplorer } from './data';
+import { EtherscanNode, InfuraNode, RPCNode, Web3Node } from 'libs/nodes';
+import { networkIdToName } from 'libs/values';
+import {
+  ETH_DEFAULT,
+  ETH_TREZOR,
+  ETH_LEDGER,
+  ETC_LEDGER,
+  ETC_TREZOR,
+  ETH_TESTNET,
+  EXP_DEFAULT,
+  UBIQ_DEFAULT,
+  DPath
+} from 'config/dpaths';
+
+export interface BlockExplorerConfig {
+  name: NetworkKeys | string;
+  tx(txHash: string): string;
+  address(address: string): string;
+}
+
+export interface Token {
+  address: string;
+  symbol: string;
+  decimal: number;
+  error?: string | null;
+}
+
+export interface NetworkContract {
+  name: NetworkKeys | string;
+  address?: string;
+  abi: string;
+}
+
+export interface DPathFormats {
+  trezor: DPath;
+  ledgerNanoS: DPath;
+  mnemonicPhrase: DPath[];
+}
+
+export interface NetworkConfig {
+  // TODO really try not to allow strings due to custom networks
+  name: NetworkKeys | string;
```

Comment:
We could type this as `NetworkKeys` if it wasn't for `makeNetworkConfigFromCustomConfig` - would appreciate feedback on how we could address this.

Reply:
I don't think there's a way around this. The custom configs are a pain for typing in general (See also, custom nodes.) We could make a specific type for custom network keys that is just string, but to be more specific about when we expect it to be either / or.